### PR TITLE
Updated kernel to 4.5 with Mali.R6P0 drivers

### PR DESCRIPTION
**Pull request by zoltan-ongithub**
The updated Kernel enables the latest Mali drivers.
The Kernel recipes are cleaned up so it is
more alligned with the structure we have on the
Dragonboard.

### CODE REVIEW

File: conf/machine/hikey.conf [hunk 1]
```diff
@@ -37,3 +42,4 @@ CMDLINE = "console=ttyAMA0,115200n8 root=/dev/mmcblk1p2 rootwait ro maxcpus=8 ea
 IMAGE_FSTYPES_append = " ext4"
 
 EXTRA_IMAGEDEPENDS = "edk2 l-loader"
+PREFERRED_VERSION_mesa = "11.%"
```
Comment (ndechesne) [inline conf/machine/hikey.conf]: mesa 11 is not in OE core, yet. we will wait for it to be merged before we can merge this change.

File: conf/machine/hikey.conf [hunk 2]
```diff
@@ -17,6 +17,11 @@ MACHINE_FEATURES = "usbhost usbgadget alsa screen wifi bluetooth efi"
 
 PREFERRED_PROVIDER_virtual/kernel ?= "linux-hikey"
 
+PREFERRED_PROVIDER_virtual/libgles1 ?= "mali450-userland-drm"
+PREFERRED_PROVIDER_virtual/libgles2 ?= "mali450-userland-drm"
+PREFERRED_PROVIDER_virtual/egl ?= "mali450-userland-drm"
```
Comment (ndechesne) [inline conf/machine/hikey.conf]: where are the mali450 recipes? not in BSP layer, i am afraid. how are we supposed to build?

File: conf/machine/hikey.conf (hunk 1 above)
Comment (koenkooi) [inline conf/machine/hikey.conf]: and MACHINEs aren't allowed to select versions, only DISTROs may do that.

File: recipes-kernel/linux/linux-hikey_4.5.bb [hunk 3]
```diff
@@ -0,0 +1,17 @@
+require linux-hikey.inc
+
+DESCRIPTION = "Hisilicon 4.5 Kernel"
+
+PV = "4.5+git${SRCPV}"
+SRCREV_kernel="${AUTOREV}"
```
Comment (ndechesne) [inline recipes-kernel/linux/linux-hikey_4.5.bb]: i don't think we want AUTOREV. it creates all kind of issues. the first issue being the inablity to reproduce builds. i will let koen comment more on that..

File: recipes-kernel/linux/linux-hikey_4.5.bb [hunk 4]
```diff
@@ -0,0 +1,17 @@
...
+
+SRC_URI = "git://github.com/xin3liang/linux;branch=hikey-tracking-integration-devel-drm;name=kernel \
+           file://defconfig  \
```
Comment (ndechesne) [inline recipes-kernel/linux/linux-hikey_4.5.bb:10]: you didn't change defconfig, though you are moving the kernel from 3.18 to 4.5. this is suspicious. is that really what you want?

File: recipes-kernel/linux/linux-hikey_4.5.bb (hunk 3 above)
Comment (koenkooi) [inline recipes-kernel/linux/linux-hikey_4.5.bb]: No AUTOREV allowed, period. Also, if you're doing AUTOREV, why set a PV? It will very soon be wrong.

**Comment by ndechesne** [comment]
also, please add signed-off-by tag in your commits.

**Comment by zoltan-ongithub** [comment]
Thanks for the review. I will update the patch as requested. I forgot to add the defconfig file..

File: recipes-kernel/linux/linux-hikey_4.5.bb (hunk 4 above)
Comment (zoltan-ongithub) [inline recipes-kernel/linux/linux-hikey_4.5.bb:10]: Defconfig is actually part of this patch, the change is just too large to be displayed by the github website.

File: conf/machine/hikey.conf (hunk 2 above)
Comment (zoltan-ongithub) [inline conf/machine/hikey.conf]: I pushed the recipe into our meta-los layer, moving this code to meta-los as well for now.

**Comment by koenkooi** [comment]
FWIW, we agreed with the 96board folks to use the 4.4 kernel, does that also work with MALI? The goal is to use the same kernel (tree) for debian and OE.

File: conf/machine/hikey.conf (hunk 2 above)
Comment (fboudra) [inline conf/machine/hikey.conf]: > I pushed the recipe into our meta-los layer, moving this code to meta-los as well for now.

why?

**Comment by zoltan-ongithub** [comment]
It may be doable but there will be some backporting work on the DRM/KMS hikey drivers. There are also few changes in the DRM/KMS kernel framework. With Weston/DRM/KMS we had some issues on earlier Kernels which where not affecting the X11 based builds: lack of vblank support in the DRM driver, problem with async page flip and lack of cursor implementation.

File: conf/machine/hikey.conf (hunk 2 above)
Comment (zoltan-ongithub) [inline conf/machine/hikey.conf]: It requires a binary blob not available for distribution yet. See:
https://github.com/kuscsik/meta-los/blob/hikey/recipes-graphics/mali-userland/mali450-userland-drm_binary.bb

**Comment by zoltan-ongithub** [comment]
Issues resolved in 9470520

**Comment by koenkooi** [comment]
"Resolved reviewer comments" is not a proper commit message. Have a look at http://www.openembedded.org/wiki/Commit_Patch_Message_Guidelines

**Comment by zoltan-ongithub** [comment]
Do you prefer squashed and a generate a new pull request?

**Comment by koenkooi** [comment]
Either way is fine

**Comment by zoltan-ongithub** [comment]
Squashed.

**Comment by koenkooi** [comment]
"Updated kernel to 4.5 with Mali.R6P0 drivers" is also not a valid commit message, have a look at http://www.openembedded.org/wiki/Commit_Patch_Message_Guidelines

And if you're touching a machine.conf, those change need to go into a seperate commit. Also, the tense to use is 'update', not 'updated' in OE commit messages. The wiki above should cover that.

**Comment by zoltan-ongithub** [comment]
Updated

**Comment by koenkooi** [comment]
The defconfig is the result of 'make savedefconfig', not a copy of .config. Can you update it?

File: recipes-kernel/linux/linux-hikey_4.5.bb [hunk 5]
```diff
@@ -0,0 +1,18 @@
+require linux-hikey.inc
+
+DESCRIPTION = "Hisilicon 4.5 Kernel"
+
+PV = "4.5+git${SRCPV}"
+
+SRCREV_kernel="623d1d7bbd697e5793a135d5334e02c0e9889ba1"
+
+SRC_URI = "git://github.com/xin3liang/linux;branch=hikey-tracking-integration-devel-drm;name=kernel \
+           file://defconfig  \
```
Comment (rsalveti) [inline recipes-kernel/linux/linux-hikey_4.5.bb:10]: While this kernel is targeted for hikey, we might want to rename the recipe as this is actually coming from a personal repository, and not really generic.

Something like linux-hikey-devel-drm-4.5 might be better.

**Comment by rsalveti** [comment]
Fine to have it 4.5 atm, we can then compare and see if we can also get it supported in the common 4.4 kernel later (since that would be a better target).

File: recipes-kernel/linux/linux-hikey_4.5.bb (hunk 5 above)
Comment (fboudra) [inline recipes-kernel/linux/linux-hikey_4.5.bb:10]: We probably don't want to use this tree but the mainline-rebase branch. This topic branch is meant to land there.

Comment (zoltan-ongithub) [inline recipes-kernel/linux/linux-hikey_4.5.bb:10]: Ok, closing pull request.

**Pull request closed without merging**